x-pack/plugins/case/server/routes/api/cases/configure/utils.ts

```diff
@@ -0,0 +1,237 @@
+  ServiceConnectorCommentParams,
+  ConnectorMappingsAttributes,
+  ConnectorTypes,
+  EntityInformation,
+  ExternalServiceParams,
+  Incident,
+  JiraPushToServiceApiParams,
+  MapIncident,
+  PipedField,
+  PrepareFieldsForTransformArgs,
+  PushToServiceApiParams,
+  ResilientPushToServiceApiParams,
+  ServiceNowPushToServiceApiParams,
+  SimpleComment,
+  Transformer,
+  TransformerArgs,
+  TransformFieldsArgs,
+} from '../../../../../common/api';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { ActionsClient } from '../../../../../../actions/server/actions_client';
+
+export const mapIncident = async (
+  actionsClient: ActionsClient,
+  connectorId: string,
+  connectorType: string,
+  mappings: ConnectorMappingsAttributes[],
+  params: ServiceConnectorCaseParams
+): Promise<MapIncident> => {
+  const { comments: caseComments, externalId } = params;
+  const defaultPipes = externalId ? ['informationUpdated'] : ['informationCreated'];
+  let currentIncident: ExternalServiceParams | undefined;
+  const service = serviceFormatter(connectorType, params);
+  if (service == null) {
+    throw new Error(`Invalid service`);
+  }
+  const thirdPartyName = service.thirdPartyName;
+  let incident: Partial<PushToServiceApiParams['incident']> = service.incident;
+  if (externalId) {
+    try {
+      currentIncident = ((await actionsClient.execute({
+        actionId: connectorId,
+        params: {
+          subAction: 'getIncident',
+          subActionParams: { externalId },
+        },
+      })) as unknown) as ExternalServiceParams | undefined;
+    } catch (ex) {
+      throw new Error(
+        `Retrieving Incident by id ${externalId} from ${thirdPartyName} failed with exception: ${ex}`
+      );
+    }
+  }
+
+  const fields = prepareFieldsForTransformation({
+    defaultPipes,
+    mappings,
+    params,
+  });
+
+  const transformedFields = transformFields<
+    ServiceConnectorCaseParams,
+    ExternalServiceParams,
+    Incident
+  >({
+    params,
+    fields,
+    currentIncident,
+  });
+  incident = { ...incident, ...transformedFields, externalId };
+  let comments: SimpleComment[] = [];
+  if (caseComments && Array.isArray(caseComments) && caseComments.length > 0) {
+    const commentsMapping = mappings.find((m) => m.source === 'comments');
+    if (commentsMapping?.action_type !== 'nothing') {
+      comments = transformComments(caseComments, ['informationAdded']);
+    }
+  }
+  return { incident, comments };
+};
+
+export const serviceFormatter = (
+  connectorType: string,
+  params: unknown
+): { thirdPartyName: string; incident: Partial<PushToServiceApiParams['incident']> } | null => {
+  switch (connectorType) {
+    case ConnectorTypes.jira:
+      const {
+        priority,
+        labels,
+        issueType,
+        parent,
+      } = params as JiraPushToServiceApiParams['incident'];
+      return {
+        incident: { priority, labels, issueType, parent },
+        thirdPartyName: 'Jira',
```

Comment:
this is (part) of the end of the missed migration @elastic/security-threat-hunting